node_modules/oae-principals/lib/internal/dao.js

```diff
@@ -347,6 +422,28 @@ var setAdmin = module.exports.setAdmin = function(adminType, isAdmin, userId, ca
 };
 
 /**
+ * Get a user by their email address. If no user could be found for a given email address,
+ * a 404 error will be returned
+ *
+ * @param  {String}     email           The email address that is associated with a user
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error object, if any
+ * @param  {User}       callback.user   The user that was associated with the email
+ */
+var getUserByEmail = module.exports.getUserByEmail = function(email, callback) {
+    Cassandra.runQuery('SELECT * FROM "PrincipalsByEmail" WHERE "email" = ?', [email], function(err, rows) {
+        if (err) {
+            return callback(err);
+        } else if (_.isEmpty(rows)) {
+            return callback({'code': 404, 'msg': 'No user found by that email'});
+        }
+
+        var principalId = rows[0].get('principalId').value;
+        return getPrincipal(principalId, callback);
+    });
+};
```

Comment:
That's what I did in the follow-up PR.. This PR only addresses the issues you've raised before we decided to take another approach. Maybe I should just close this one, and submit it as one PR?